### PR TITLE
folder str

### PR DESCRIPTION
**Pull request by imsandip2085**

### CODE REVIEW

File: src/component/form/Form.jsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import DatePicker from "react-date-picker";
+
+
+
+function TodoForm () {
+    const [title, setTitle] = useState("");
+    const [todoItem, setTodoItem] = useState([]); 
+    const [date, setDate] = useState(new Date());
+
+    console.log(todoItem,"dddddddddddddddd");
```
Comment (saurabhketech) [inline src/component/form/Form.jsx:16]: remove log, this will never comes in pr

File: src/component/form/Form.jsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
+import React, {useState} from 'react';
```
Comment (saurabhketech) [inline src/component/form/Form.jsx:1]: code must be formatted

File: src/component/form/Form.jsx (hunk 1 above)
Comment (imsandip2085) [reply src/component/form/Form.jsx:16]: ok sir

File: src/component/form/Form.jsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+import SubmitButton from '../submitButton/SubmitButton';
+// import { useState } from 'react';
+import { CardContent, Card, } from "@material-ui/core";
+import { makeStyles } from "@material-ui/core/styles";
+import DatePicker from "react-date-picker";
+
+
+
+function TodoForm () {
+    const [title, setTitle] = useState("");
```
Comment (saurabhketech) [inline src/component/form/Form.jsx:12]: dont take too many states, use single state as i told you

File: src/component/form/Form.jsx (hunk 2 above)
Comment (imsandip2085) [reply src/component/form/Form.jsx:1]: ok sir

File: src/component/filterButton/ascButton/AscButton.jsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import React from "react";
+import { Button } from "@material-ui/core";
+import { makeStyles } from "@material-ui/core/styles";
+
+function AscButton(props) {
```
Comment (manishiitg) [inline src/component/filterButton/ascButton/AscButton.jsx:5]: this is wrong pattern. 
we making duplicates of the same button.

this is also wrong. if not clear ask me @saurabhketech

File: src/pages/home/Home.jsx [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+function Home() {
+  const [formState, changeFormStates] = useState({
+    title: "",
+    date: new Date(),
+    todoItem: [],
+    buttonStatus: "",
+  });
+
+  const handleSubmit = (e) => {
```
Comment (manishiitg) [inline src/pages/home/Home.jsx:14]: make this to a custom hook. as per my blogs. all this code should not be here at all

File: src/customHook/CustomHook.jsx [hunk 6]
```diff
@@ -0,0 +1,79 @@
+import React from "react";
+
+function useCustomHooks() {
```
Comment (manishiitg) [inline src/customHook/CustomHook.jsx:3]: this is wrong . the name of fucntion should not be custom hook. 

it should be as per the state. like useTodoForm something liek this.

File: src/component/todoList/homeTodoList/HomeTodoList.jsx [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+  return (
+    <div className={classes.todolistItems}>
+      {props.formState.buttonStatus == "all" ? (
+        <AllTodoList
+          formStates={props.formState}
+          handleChecked={(id) => props.handleChecked(id)}
+          handleDeleteListItem={(id) => props.handleDeleteListItem(id)}
+        />
+      ) : props.formState.buttonStatus == "completed" ? (
+        <CompletedTodoList
```
Comment (manishiitg) [inline src/component/todoList/homeTodoList/HomeTodoList.jsx:49]: this structure is also fully wrong.

why do we have so many different view for the same thing. 

 we should just changing data and have a single view

File: src/pages/home/Home.jsx (hunk 5 above)
Comment (imsandip2085) [reply src/pages/home/Home.jsx:14]: yes sir

File: src/customHook/CustomHook.jsx (hunk 6 above)
Comment (imsandip2085) [reply src/customHook/CustomHook.jsx:3]: ok sir,

File: src/component/todoList/homeTodoList/HomeTodoList.jsx (hunk 7 above)
Comment (imsandip2085) [reply src/component/todoList/homeTodoList/HomeTodoList.jsx:49]: ok sir

File: src/component/filterButton/ascButton/AscButton.jsx (hunk 4 above)
Comment (saurabhketech) [reply src/component/filterButton/ascButton/AscButton.jsx:5]: @imsandip2085 this is wrong i already explained you need use one component for each button will be single component for all pages

File: src/component/filterButton/completed/Completed.jsx [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+function CompletedButton(props) {
+  const useStyles = makeStyles((theme) => ({
+    boxs: {
+      display: "inline-block",
+    },
+  }));
+  const classes = useStyles();
+
+  return (
+    <div className={classes.boxs}>
```
Comment (saurabhketech) [inline src/component/filterButton/completed/Completed.jsx:14]: same here

**Pull request merged**